src/wovnio/wovnphp/Headers.php

```diff
@@ -19,18 +19,23 @@ class Headers
     private $urlLang;
     private $query;
     private $browserLang;
+    private $cookies;
+    private $cookieLang;
 
     /**
      * Constructor of the Headers class
      *
      * @param array &$env Contains the _SERVER env variable
      * @param Store &$store The store containing user settings
+     * @param array $cookies &$cookies Contains the _COOKIE env variable
      * @return void
      */
-    public function __construct(&$env, &$store)
+    public function __construct(&$env, &$store, $cookies)
     {
         $this->env =& $env;
         $this->store =& $store;
+        $this->cookies = $cookies;
+        $this->cookieLang = new CookieLang($this);
```

Comment:
I think WOVN.php doesn't need to keep all cookies.
So, my suggestion is remove `$this->cookies = $cookies;`.
And set `$cookies` to `CookieLang` and only keep `$this->cookieLang` .

Reply:
The reason I kept a separate copy of cookies (which came from the superglobal $_COOKIE, is to facilitate testing, so that we can easily mock a cookie value. The same way $env is used.

Reply:
I would also argue Headers does not need a  copy of $_COOKIE at all. Headers only has it to give to CookieLang. Just give it to CookieLang. Which would also solve your dependency issue.

Reply:
Looks good. I think it doesn't use Headers anymore though thus can be simplified further like I mentioned.

Reply:
Good catch. I must be blind.